src/main/java/core/basesyntax/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;

```

Comment:
The field `shopServiceImpl` is not initialized. It's better to use a constructor to initialize it.